packages/datadog-plugin-express/test/index.spec.js

```diff
@@ -714,7 +715,7 @@ describe('Plugin', () => {
 
           app.get('/user', (req, res) => {
             res.statusCode = 400
-            throw new Error('boom')
+            res.status(400).send()
```

Comment:
Why this change?

Reply:
Explicitly throwing an error in the middleware, when middleware instrumentation is disabled, gets picked up regardless of the statusCode due to the additional logic introduced by this PR.

I had considered also doing some work within the `reactivateAndWrapMiddlewareErrors` (yea, the name is gross, sorry, open to suggestions) method so that it only _conditionally_ marks a span as an error based on `validateStatus` but it's not clear to me that that the response's statusCode will always be set intentionally within the middleware. 

As this test is meant to test status Code handling it made sense to me to just simplify this test. This seems to be the only test this sortve edge case really touches but I suppose I am introducing a slightly new behavior here in the situation where a user has both disabled middleware instrumentation and wants to return a non-error status code while throwing an error in the middleware? 

Lmk what you think, I also may have explained that poorly so please lmk if i can clarify anything.

Reply:
Still not sure I understand. The reason for throwing in this test was that throwing is a different code path than responding. In any case, the status code should be handled properly at the HTTP level since if the error is unhandled it will reach the request, and if it's handled then it will be the user decision.

Reply:
Ok, I think I misunderstood something here. In the other test case for `should only handle errors for configured status codes` There is still an error being marked, however it's on the middleware span. In this case we will still need to mark an error, but the request span is the active scope so it needs to be marked there vs the middleware (which we are not tracking separately with middleware tracing disabled). In this case I've also updated the spec test to be explicit about the behavior `should mark middleware errors regardless of status codes configuration`